sdk/storage/azure-storage-blob-nio/src/main/java/com/azure/storage/blob/nio/AzureFileSystemProvider.java

```diff
@@ -1159,4 +1195,35 @@ private String extractAccountEndpoint(URI uri) {
 
         return endpoint;
     }
+
+    /**
+     * Sets the default configurations for this {@code FileSystemProvider}. The configurations will be applied if a call
+     * to {@link Paths#get(URI)} is made and there is not an open file system pointing to the specified endpoint. In
+     * this case, a corresponding file system will be opened. Otherwise, the already opened file system will be used.
+     * The endpoint must be specified in the uri.
+     *
+     * A file system will only be opened automatically if these defaults are set. If they are not set, attempting to
+     * retrieve a path in an unopened file system will continue to throw a {@link FileSystemNotFoundException}.
+     *
+     * In the case where an endpoint is not set on the uri, the default endpoint provided here will be used. If the
+     * default endpoint is set and an endpoint is provided on the uri, the one provided on the uri will be preferred.
+     *
+     * @param config The configurations map. Please see the docs on {@link AzureFileSystemProvider for more information}
+     * @param endpoint The default endpoint to set. May be null
+     * @throws IllegalStateException If the configurations have already been set
+     */
+    public static void setDefaultConfigurations(Map<String, Object> config, String endpoint) {
```

Comment:
Looks like we extracted endpoint for convenience. Which seems to indicate that `Map<String, Object>` is not easy to use, i.e. it doesn't describe very well types and properties allowed.
Since this API isn't related to any NIO SPI then maybe we should consider taking here strongly typed options instead of map and convert it into Map<String,Object> behind the scenes?

Reply:
Without making any claims about whether the map is easy to use or not as that can be a separate discussion, we extracted the endpoint for a different reason than for convenience. Typically, customers will create a new file system via `FileSystemProvider#newFileSystem(uri, map)`. The idea was to imitate this signature but allow access to the behavior in a different location (though looking at it, I should potentially swap the order). The ask for a default endpoint came later in the discussion, and I didn't want to add it into the map because we don't respect that map option in the `newFileSystem()` method as that's not how it's intended to be used. It seemed like it would be confusing if the map options were identical except for the endpoint option which was respected in one method but ignored in another.